lib/src/background_usecase.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter_clean_architecture/flutter_clean_architecture.dart';
+import 'dart:isolate';
+import 'package:meta/meta.dart';
+import 'package:rxdart/rxdart.dart';
+import 'dart:async';
+
+enum BackgroundUseCaseState { idle, loading, calculating }
+typedef UseCaseTask = void Function(BackgroundUseCaseParams message);
```

Comment:
Shouldn't be better to use a pattern to define this parameter?

```suggestion
typedef UseCaseTask = void Function(BackgroundUseCaseParams backgroundUseCaseParams);
```

or just:

```suggestion
typedef UseCaseTask = void Function(BackgroundUseCaseParams params);
```

Just `message` seems a little confusing for me